server/src/main/java/org/opensearch/search/aggregations/AggregatorBase.java

```diff
@@ -216,6 +220,10 @@ protected void preGetSubLeafCollectors(LeafReaderContext ctx) throws IOException
      */
     protected void doPreCollection() throws IOException {}
 
+    protected boolean tryPrecomputeAggregationForLeaf(LeafReaderContext ctx) throws IOException {
```

Comment:
Are we planning to add support for subAggregations later ?

Don't we need something similar to LeafBucketCollector [ mainly the collect equivalent method ] - where the subAggregators can return the implementation and the parent class can invoke the subAggs implementation to collect each of the unique values/buckets.

[ This requirement is mainly for star tree aggregations ]
```
For example : for Terms aggs with sum sub-aggs,

Terms |  sum
1            500    --> Bucket 1
2           6000   --> Bucket 2
```

Reply:
I was thinking that it would be the responsibility of an `Aggregator` to delegate to its subaggregators. 

I think @sandeshkr419 managed to do that in https://github.com/opensearch-project/OpenSearch/pull/16674